### PR TITLE
Fix for #365

### PR DESCRIPTION
**Pull request by SergeyTeplyakov**

### CODE REVIEW

File: Foxtrot/Driver/Rewriting/EmitAsyncClosure.cs [hunk 1]
```diff
@@ -195,6 +195,7 @@ public TypeNode GetEnclosingTypeParameterByClosureTypeParameter(TypeNode closure
         private readonly Cache<TypeNode> func2Type;
 
         private readonly Dictionary<Local, MemberBinding> closureLocals = new Dictionary<Local, MemberBinding>();
+        private readonly Dictionary<MemberBinding, MemberBinding> closureMembers = new Dictionary<MemberBinding, MemberBinding>();
```
Comment (SergeyTeplyakov) [inline Foxtrot/Driver/Rewriting/EmitAsyncClosure.cs]: Removed this line.

**Comment by SergeyTeplyakov** [comment]
Guys, will appreciate if you'll look at my changes.

**Comment by yaakov-h** [comment]
I don't know the rewriter well but the changes seem to make sense.

Does this handle odd cases like `Contract.Ensures(Contract.ForAll(_collection, o => o != null))` or `Contract.Ensures(Contract.ForAll(Contract.Result<IEnumerable<int>>(), o => o > _someVar))`?

**Comment by hubuk** [comment]
Looks fine.

**Comment by SergeyTeplyakov** [comment]
@yaakov-h Yep, this should be covered as well. I think I even saw such kind of test cases when I worked on this (because once I've broke some of them).

**Pull request merged**